CMakeLists.txt

```diff
@@ -256,15 +254,13 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   # CUDA does not support current Clang as host compiler, we need use gcc
   # CMAKE_CUDA_HOST_COMPILER variable operates on paths
   set(CUDA_UNSUPPORTED_COMPILER 0)
-  if (NOT DALI_CLANG_ONLY)
+  if (NOT DALI_CLANG_ONLY AND NOT CMAKE_CUDA_HOST_COMPILER)
```

Comment:
I think that the condition is OK - however, it should probably be preceded by:
```cmake
if (NOT CMAKE_CUDA_HOST_COMPILER)
  set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_C_COMPILER})
endif()
```
@klecki What's your take on this?

Reply:
Some of this is probably legacy. I can even use clang (up to 9.0) with CUDA 11.0 as host compiler apparently (for example).

I need to run few experiments with it, will do so in some spare time, maybe we can remove this and the next if completely, and let the user decide now.

Reply:
@cyyever What purpose does this change serve? Do you need this for something? It's CMake codepath for non-standard build.